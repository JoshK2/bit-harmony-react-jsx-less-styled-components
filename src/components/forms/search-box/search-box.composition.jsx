import React from 'react';
import { SearchBox } from './search-box';

export const SearchBoxExample = () => {
  return <SearchBox />;
};
