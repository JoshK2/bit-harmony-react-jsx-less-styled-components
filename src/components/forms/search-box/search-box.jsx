import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Input } from '../../base-ui/input';
import { Button } from '../../base-ui/button';

export const SearchBox = ({ onSubmit }) => {
  const [value, setvalue] = useState('');

  return (
    <form>
      <Input
        placeholder="Search your movies"
        value={value}
        onChange={setvalue}
      />
      <Button onClick={onSubmit}>Search</Button>
    </form>
  );
};

SearchBox.propTypes = {
  // Submit the research
  onSubmit: PropTypes.func,
};

SearchBox.defaultProps = {
  onSubmit: () => alert('Click on Search!'),
};
