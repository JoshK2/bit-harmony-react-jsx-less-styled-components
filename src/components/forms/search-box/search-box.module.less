.main {
  padding: 8px 10px;
  display: flex;
  flex-direction: row;
  height: 50px;
}
.input {
  margin-right: 5px;
}
.button {
  margin-left: 5px;
}
