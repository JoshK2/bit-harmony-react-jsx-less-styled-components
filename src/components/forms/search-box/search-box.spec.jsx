import React from 'react';
import testRenderer from 'react-test-renderer';
import { SearchBoxExample } from './search-box.composition';

describe('Search Box component', () => {
  it('renders correctly', () => {
    testRenderer.create(<SearchBoxExample />);
  });
  it('snapshot test', () => {
    const component = testRenderer.create(<SearchBoxExample />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
