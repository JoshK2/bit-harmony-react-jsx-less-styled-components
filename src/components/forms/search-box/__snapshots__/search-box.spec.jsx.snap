// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box component snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="sc-bdfBwQ jyVkcv"
    placeholder="Search your movies"
  />
  <button
    className="sc-gsTCUz fTGdlx"
    type="submit"
  >
    Search
  </button>
</form>
`;
